packages/dbos-cloud/databases/databases.ts

```diff
@@ -377,3 +386,70 @@ export async function connect(host: string, dbName: string, password: string) {
     return 1;
   }
 }
+
+export async function chooseAppDBServer(logger: Logger, host: string, userCredentials: DBOSCloudCredentials, userDBName: string = ""): Promise<string> {
+  // List existing database instances.
+  let userDBs: UserDBInstance[] = [];
+  const bearerToken = "Bearer " + userCredentials.token;
+  try {
+    const res = await axios.get(`https://${host}/v1alpha1/${userCredentials.organization}/databases`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: bearerToken,
+      },
+    });
+    userDBs = res.data as UserDBInstance[];
+  } catch (e) {
+    const errorLabel = `Failed to list databases`;
+    const axiosError = e as AxiosError;
+    if (isCloudAPIErrorResponse(axiosError.response?.data)) {
+      handleAPIErrors(errorLabel, axiosError);
+    } else {
+      logger.error(`${errorLabel}: ${(e as Error).message}`);
+    }
+    return "";
+  }
+
+  if (userDBName) {
+    // Check if the database instance exists or not.
+    const dbExists = userDBs.some((db) => db.PostgresInstanceName === userDBName);
+    if (dbExists) {
+      return userDBName;
+    }
+  }
+
+  if (userDBName || userDBs.length === 0) {
+    // If not, prompt the user to provision one.
+    if (!userDBName) {
+      logger.info("No database found, provisioning a database instance (server)...");
+      userDBName = await input({
+        message: "Database instance name?",
+        default: `${userCredentials.userName}-db-server`,
+      });
+    } else {
+      logger.info(`Database instance ${userDBName} not found, provisioning a new one...`);
+    }
+
+    // Use a default user name and auto generated password.
+    const appDBUserName = "dbos_user";
+    const appDBPassword = Buffer.from(Math.random().toString()).toString("base64");
+    const res = await createUserDb(host, userDBName, appDBUserName, appDBPassword, true);
+    if (res !== 0) {
+      return "";
+    }
+  } else if (userDBs.length > 1) {
+    // If there is more than one database instances, prompt the user to select one.
+    userDBName = await select({
+      message: "Choose a database instance for this app:",
+      choices: userDBs.map((db) => ({
+        name: db.PostgresInstanceName,
+        value: db.PostgresInstanceName,
```

Comment:
Maybe the value prompt should also print if that's a linked database or not (supabase, etc)?